mlflow/utils/request_utils.py

```diff
@@ -41,9 +42,10 @@ def augmented_raise_for_status(response):
 
 def download_chunk(range_start, range_end, headers, download_path, http_uri):
     combined_headers = {**headers, "Range": f"bytes={range_start}-{range_end}"}
+    timeout = DOWNLOAD_CHUNK_TIMEOUT_100M * ((range_end - range_start) // 100_000_000 + 1)
```

Comment:
Why not use `DOWNLOAD_CHUNK_TIMEOUT_100M * CHUNK_SIZE / /  100_000_000` ?

Reply:
chunk_size is not the same for databricks_artifact_repo and databricks_models_artifact_repo, but they're passed such that chunk_size==range_end-range_start+1

Reply:
If (range_end - range_start) might be a small value (e.g. the tail chunk), we should set a lower bound for timeout, for small chunk the time is mainly determined by network latency.

Reply:
In that case, maybe we just use the float number for timeout: DOWNLOAD_CHUNK_TIMEOUT_100M * (range_end - range_start) / 100_000_000 ?

Reply:
Do we need a test case for this? Though we could only mock the request result

Reply:
> In that case, maybe we just use the float number for timeout: DOWNLOAD_CHUNK_TIMEOUT_100M * (range_end - range_start) / 100_000_000 ?

NO. using float does not address it.

e.g., if (range_end - range_start)== 1, this way it gets timeout of 1e-06 second, it is too short.

I think we need logic like:

```

timeout = max(
  DOWNLOAD_CHUNK_TIMEOUT_100M * (range_end - range_start) / 100_000_000,
  5,  # lower bound of timeout
)

```